### DISCUSSION TITLE
conservative interpolation from global regular lat/lon to rotated pole grid_lat/grid_lon

### DISCUSSION THREAD

**Discussion by hellovyu** [SciTools/iris#6009] [Q&A]
Hi,
Just wondered if it's possible to perform conservative interpolation for global grid with regular latitude/longitude onto regional rotated pole grid with grid_latitude/grid_longitude with iris?

I just had errors when trying
`ocube = globalcube.regrid(regionalgrid, iris.analysis.AreaWeighted())`
error: The horizontal grid coordinates of both the source and grid cubes must have the same coordinate system.

`print(regionalgrid.coord_system())`
RotatedGeogCS(39.59999847412109, 195.0, ellipsoid=GeogCS(6371229.0))
`print(globalgrid.coord_system())`
 GeogCS(6371229.0)

I found this description on [iris website](https://scitools-iris.readthedocs.io/en/latest/userguide/interpolation_and_regridding.html):
One of the key limitations of the AreaWeighted regridding scheme is that the two input grids must be defined in the same coordinate system as each other. Both input grids must also contain monotonic, bounded, 1D spatial coordinates.

So my question was: is this possible to convert global grid to the regional rotated pole grid to make iris.analysis.AreaWeighted() works? or any suggestions on doing this conservative interpolation? Thanks!

**Comment by trexfeathers** [comment]
From @SciTools/peloton:

Hi @hellovyu, thanks for getting in touch. As you've noticed, conservative regridding does not work here. Other forms of conversion (i.e. not regridding) would make the data meaningless, so that's not an option. We suggest you look at iris-esmf-regrid, which uses spherical mathematics and so coordinate systems are not a problem there. [Read more here](https://scitools-iris.readthedocs.io/en/stable/further_topics/which_regridder_to_use.html#)

**Reply by hellovyu** [reply]
Thanks @trexfeathers and @SciTools/peloton for prompt reply!
I tried ESMFAreaWeighted as you suggested, it worked for the two different coordinate systems! 
But when I compared the sums of the ESMFAreaWeighted regrided data with the global data over the corresponding region, the two sums are not comparable. And the means are similar comparing the Linear regrided data and ESMFAreaWeighted regrided data.

So I think it looks like the ESMFAreaWeighted method didn't give me conservetive regrided result and it regrided data in a similar way as the Linear method?
And the description says [The property which these regridders conserve is the global area weighted average of the data](https://scitools-iris.readthedocs.io/en/stable/further_topics/which_regridder_to_use.html#). So I was wondering if the non-conservative issue I have with my data is because my source data is global and my target is regional? 
Thanks in advance for your help!

**Reply by stephenworsley** [reply]
What exactly do you mean by "comparing the sums of the ESMFAreaWeighted regrided data with the global data over the corresponding region"? I would expect that taking a sum of data over a region will not always give you a meaningful value and so you should not always expect these values to match. For example, if you were looking at two pieces of data which both described temperature over a given region, but one was described on a finer grid with more points than the other, then the sum of the data on the fine grid would naturally be greater than the sum of the data for data on the coarser grid simply because there were more data points to add. On the other hand, It's possible that the data could represent something like "total rainfall over the given grid box", in which case the sum of data would be meaningful, however, such data would be unsuitable for traditional regridding.
It may be that an area-weighted sum or area-weighted average would be a more suitable quantity to compare. to determine if the regridding is valid.

**Comment by hellovyu** [comment]
Thanks for the previous comments and help! I learnt a lot and am having a second try to clarify my question:
My source data is a global distrubution of mass on regular lat/lon grid.
My target grid is a regional rotated pole grid.
My aim is to regrid my global source data onto the regional rotated role grid in a conservative way.

I came across a similar question https://groups.google.com/g/scitools-iris/c/m5B2752zRjQ
The possible solution mentioned in the post was: 
find the 2D regular lat&lon for the rotated pole grid points -> '_regrid/reproject to move the actual data points onto a regular grid in lat/lon space_' -> area-weighted regriding
I then have to take one step further to re-rotate the lat/lon grid back onto the rotated pole grid because my final aim is to get data on the target rotated grid.

So I was now wondering 1) if it's technically possible to do the last bit of re-rotating the lat/lon global grid onto the target rotated pole grid with iris? I know there is iris.analysis.cartography.rotate_pole, I guess this will do the work? 
2) Because I suppose lat/lon intervals among the unrotated grid cells should be different. So I am bit worried that the regridding of the rotated pole grid onto regular lat/lon grid will cause inaccuracy to some extent.
3) Because of the concern in 2), I was wondering if I could write a function myself to estimate the varing lat/lon intervals and calculate the area-weighted values after I get the real lat/lon of the rotated grid cells. This sounds easier because I don't have to project the unrotated lat/lon points onto regular lat/lon grid and regrid it back onto regional rotated grid.

Thanks in adcance for any of your suggestions and help!

**Reply by trexfeathers** [reply]
Hi @hellovyu,

The Google Groups question you cross-reference only recommends using 2D regular lat&lon for plotting. Unfortunately conservative regridding in Iris (`AreaWeighted`) only works with orthogonal, bounded, 1D coordinates ([ref](https://scitools-iris.readthedocs.io/en/stable/further_topics/which_regridder_to_use.html#areaweighted)).

It sounds like you were disappointed with the results you got when using iris-esmf-regrid, and so you are looking for alternative workflows. I can confidently say that any workflow you can get to complete successfully will not produce a mathematically valid answer. If you look at the example below: iris-esmf-regrid is **our only tool** with the intelligence to say how red each target cell should be (EDIT: sorry, you need a white background for this to make sense!).

I echo @stephenworsley's [comment](https://github.com/SciTools/iris/discussions/6009#discussioncomment-9819664): it is important to use the correct metrics to measure the accuracy of regridding, and it may be that the result you got from iris-esmf-regrid was already the one you needed. Plotting can sometimes help when investigating, did you try that too? 

iris-esmf-regrid isn't the only way to calculate this answer, however, it's just that other ways have not been implemented - you would need to write the mathematics yourself. And of course if you do this then we would love to apply the improvement to Iris itself via a pull request!

![image](https://github.com/user-attachments/assets/665322b4-f19e-4c47-a4c8-a48608fc39ad)
